script2.js

```diff
@@ -0,0 +1,82 @@
+// Assignment Code
+var generateBtn = document.querySelector("#generate");
+
+function generatePassword() {
+  // PLEASE ADD ALL YOUR CODE HERE
+
+  //declare criteria as an object
+  const lowerCase = "abcdefghijklmnopqrstuvwxyz";
+  const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+  const numeric = "0123456789";
+  const special = "!@£$%&*?+=";
+
+  // get password length
+  const passwordLength = Number(prompt("Enter password length:"));
+
+  // if password is a number and within range of 8 and 128 do rest else alert and return ""
+
+  console.log(passwordLength);
+
+  if (passwordLength < 8 || passwordLength > 128) {
+    alert("Please choose a different length?");
+    return passwordLength;
+  } else if (passwordLength >= 8 || passwordLength <= 128) {
+  }
+
+  //get criteria const input
+  const isLowercase = confirm("Do you want lowercase?");
+  const isUppercase = confirm("Do you want uppercase?");
+  const isNumeric = confirm("Do you want numeric?");
+  const isSpecial = confirm("Do you want special?");
+
+  if (!isLowercase && !isUppercase && !isNumeric && !isSpecial) {
+    alert();
+    return;
+  }
+  //return function
+  //if statement for return
+  const choices = [];
+
+  if (isLowercase) {
+    choices.push(lowerCase);
+  }
+  if (isUppercase) {
+    choices.push(uppercase);
+  }
+
+  if (isNumeric) {
+    choices.push(numeric);
+  }
+
+  if (isSpecial) {
+    choices.push(special);
+  }
+
+  let password = "";
+
+  // generate random password
+  for (let i = 0; i < 10; i++) {
```

Comment:
The length of the password will always be 10 as you have wrote `i < 10`. Instead, you should use the password length variable here which will continue to loop while the index is less than the inputted password length.